orttraining/orttraining/python/training/onnxblock/checkpoint_utils.py

```diff
@@ -17,3 +20,16 @@ def save_checkpoint(parameters, path_to_checkpoint):
     trainable_params = [param.SerializeToString() for param in trainable_params]
     non_trainable_params = [param.SerializeToString() for param in non_trainable_params]
     _internal_save_checkpoint(trainable_params, non_trainable_params, path_to_checkpoint)
+
+
+def load_checkpoint(path_to_checkpoint, model):
+    """Loads the checkpoint to an onnx inference model."""
+
+    # Load the parameters from the checkpoint
+    parameters = _internal_load_checkpoint(path_to_checkpoint)
+
+    parameters_dict = {}
+    for (i, initializer) in enumerate(model.graph.initializer):
+        parameters_dict[initializer.name] = TensorProto()
+        parameters_dict[initializer.name].ParseFromString(parameters[i])
+        initializer.CopyFrom(parameters_dict[initializer.name])
```

Comment:
We are potentially assigning empty `TensorProto`s here since initializer.name might not be the same as parameters[i].name.

Reply:
+1 this does not look right... how do you guarantee the order is the same?